### PR TITLE
Removing boolean checks on python values for clipping

### PR DESCRIPTION
**Pull request by Neo-X**
These don't work if you want to use a tensor or variable as an argument for the clipping terms.

### CODE REVIEW

**Comment by Neo-X** [comment]
Will do.

**Comment by Neo-X** [comment]
Done.

File: tests/keras/backend/backend_test.py [hunk 1]
```diff
@@ -1833,6 +1833,16 @@ def test_variable_support_bool_dtype(self):
             with pytest.raises(TypeError):
                 K.variable('', dtype='unsupported')
 
+    def test_clip_with_tensors(self):
+        # Create tensor
+        import keras
```
Comment (fchollet) [inline tests/keras/backend/backend_test.py]: You should not be importing `keras` here. You can use `K.placeholder` to create a placeholder.

File: tests/keras/backend/backend_test.py [hunk 2]
```diff
@@ -1833,6 +1833,16 @@ def test_variable_support_bool_dtype(self):
...
+        x_placeholder = K.placeholder(ndim=0)
+        y = keras.layers.Input(shape=(1,))
+		# y_placeholder = KTF.placeholder(shape=())
```
Comment (fchollet) [inline tests/keras/backend/backend_test.py]: Please remove commented-out code.

**Comment by fchollet** [comment]
CNTK tests seem to be failing: https://travis-ci.org/keras-team/keras/jobs/367340935

**Comment by Neo-X** [comment]
I will have a look at this at some point. I have never used CNTK.

File: tests/keras/backend/backend_test.py [hunk 3]
```diff
@@ -1833,6 +1833,14 @@ def test_variable_support_bool_dtype(self):
             with pytest.raises(TypeError):
                 K.variable('', dtype='unsupported')
 
+    def test_clip_with_tensors(self):
+        # Create tensors
+        x_placeholder = K.placeholder(ndim=0)
```
Comment (joelthchao) [inline tests/keras/backend/backend_test.py]: You should provide empty shape like you did in next line. You can find the reason in the cntk backend code:
```python
def placeholder(
    # ...
    if not shape:
        if ndim:
            shape = tuple([None for _ in range(ndim)])
    # shape is None and not iterable.
    cntk_shape = [dynamic_dimension if s is None else s for s in shape]
```

File: tests/keras/backend/backend_test.py [hunk 4]
```diff
@@ -1833,6 +1833,14 @@ def test_variable_support_bool_dtype(self):
...
+        y_placeholder = K.placeholder(shape=())
+        # Check correct values
+        clip_ = K.clip(y_placeholder, x_placeholder + 1.0, x_placeholder - 1.0)
+        assert isinstance(clip_, type(x_placeholder))
+
```
Comment (joelthchao) [inline tests/keras/backend/backend_test.py]: In addition to type, I think you should also check the value after clipping by placeholder variable.

**Comment by fchollet** [comment]
Closing inactive PR.

**Pull request closed without merging**